cmake/SetupCompilerOptions.cmake

```diff
@@ -275,18 +289,22 @@ blt_append_custom_compiler_flag(
      XL       "qhalt=w"       # i info, w warning, e error, s severe (default)
      )
 
-#
-# Modify flags to avoid static linking runtime issues on windows.
-# (adapted from RAJA)
-#
 
-if ( COMPILER_FAMILY_IS_MSVC AND NOT BUILD_SHARED_LIBS )
+# Attempt to fix some windows flags
+if ( COMPILER_FAMILY_IS_MSVC )
     foreach(flag_var
             CMAKE_CXX_FLAGS CMAKE_CXX_FLAGS_DEBUG CMAKE_CXX_FLAGS_RELEASE
             CMAKE_CXX_FLAGS_MINSIZEREL CMAKE_CXX_FLAGS_RELWITHDEBINFO)
-        if(${flag_var} MATCHES "/MD")
+        
+        # Avoid static linking runtime issues on windows (adapted from RAJA)
+        if(NOT BUILD_SHARED_LIBS AND ${flag_var} MATCHES "/MD")
             string(REGEX REPLACE "/MD" "/MT" ${flag_var} "${${flag_var}}")
-        endif(${flag_var} MATCHES "/MD")
+        endif()
+
+        # Clear runtime check flags -- incompatible with optimization flags
+        if(${flag_var} MATCHES "/RTC")
```

Comment:
I was able to track this one down and it was an error on our end.  This workaround for BLT is not necessary, and I will remove it.

The problem was due to a flag in one of our tests for warnings about aliasing where we were explicitly adding the ``O2`` flag (even in debug configurations).  There was a note in our CMake setup code that this would likely need a workaround for msvc (and it does).  

I will take care of it.